From 91fe7e9ed5e644672d4da67a9b0396f545d90cb6 Mon Sep 17 00:00:00 2001
From: Tim Carney <tim.carney@cargt.com>
Date: Tue, 30 Sep 2025 08:45:53 -0500
Subject: [PATCH] Autosave env when defaults are set.

---
 env/common.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/env/common.c b/env/common.c
index 32203e66731..97217b330b5 100644
--- a/env/common.c
+++ b/env/common.c
@@ -278,6 +278,8 @@ void env_set_default(const char *s, int flags)
 
 	gd->flags |= GD_FLG_ENV_READY;
 	gd->flags |= GD_FLG_ENV_DEFAULT;
+
+	env_save();
 }
 
 
